include/llvm-dialects/Dialect/Dialect.td

```diff
@@ -69,6 +69,9 @@ class Attr<string cppType_> : MetaType {
   // A check statement that is issued before using the C++ value in builders.
   // $0 is the C++ value.
   string check = "";
+  
+  // Overriding prevents generating a setter method. Attributes are mutable by default.
+  string isImmutable = "false";
```

Comment:
TableGen has booleans...
```suggestion
  bit isImmutable = false;
```
... and knock-on changes that are required for that.